File: windows/Model/Result.cs
using System;

namespace Duck.Cameras.Windows.Model
{
    public class Result<T>
    {
        public static Result<T> Ok(T value)
        {
            Result<T> result = new Result<T>();
            result.Value = value;
            result.Success = true;
            result.Exception = null;
            return result;
        }

        public static Result<T> Error(T value, Exception exception = null)
        {
            Result<T> result = new Result<T>();
            result.Value = value;
            result.Success = false;
            result.Exception = exception;
            return result;
        }

        public bool Success { get; private set; }
        public T Value { get; private set; }
        public Exception Exception { get; private set; }
    }
}

File: android/lib/src/main/java/duck/cameras/android/model/Result.java
package duck.cameras.android.model;

public class Result<T> {
    private T value;
    private boolean status;
    private Throwable throwable;

    public static <T> Result<T> ok(T value) {
        Result<T> result = new Result<>();
        result.value = value;
        result.status = true;
        result.throwable = null;
        return result;
    }

    public static <T> Result<T> error(T value) {
        return error(value, null);
    }

    public static <T> Result<T> error(T value, Throwable throwable) {
        Result<T> result = new Result<>();
        result.value = value;
        result.status = false;
        result.throwable = throwable;
        return result;
    }

    public boolean ok() {
        return status;
    }

    public Throwable error() {
        return throwable;
    }

    public T value() {
        return value;
    }
}
